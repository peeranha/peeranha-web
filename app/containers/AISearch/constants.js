export const GET_SEARCH_RESULT = 'app/AISearch/GET_SEARCH_RESULT';
export const GET_SEARCH_RESULT_SUCCESS = 'app/AISearch/GET_SEARCH_RESULT_SUCCESS';
export const GET_CHUNK_SUCCESS = 'app/AISearch/GET_CHUNK_SUCCESS';
export const STOP_GENERATION = 'app/AISearch/STOP_GENERATION';
export const START_OVER = 'app/AISearch/START_OVER';
export const GET_SEARCH_RESULT_ERROR = 'app/AISearch/GET_SEARCH_RESULT_ERROR';
