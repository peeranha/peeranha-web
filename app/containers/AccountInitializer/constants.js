/*
 *
 * AccountInitializer constants
 *
 */

export const GET_CURRENT_ACCOUNT = 'app/AccountInitializer/GET_CURRENT_ACCOUNT';
export const GET_CURRENT_ACCOUNT_SUCCESS =
  'app/AccountInitializer/GET_CURRENT_ACCOUNT_SUCCESS';
export const GET_CURRENT_ACCOUNT_ERROR =
  'app/AccountInitializer/GET_CURRENT_ACCOUNT_ERROR';

export const SELECT_ACCOUNT = 'app/SignUp/SELECT_ACCOUNT';
export const SELECT_ACCOUNT_SUCCESS = 'app/SignUp/SELECT_ACCOUNT_SUCCESS';
export const SELECT_ACCOUNT_ERROR = 'app/SignUp/SELECT_ACCOUNT_ERROR';

export const NO_SCATTER = 'app/SignUp/NO_SCATTER';
export const NO_SELECTED_SCATTER_ACCOUNTS =
  'app/containers/Button/NO_SELECTED_SCATTER_ACCOUNTS';
export const USER_IS_ABSENT_IN_SYSTEM_AND_LOGIN =
  'app/containers/Button/USER_IS_ABSENT_IN_SYSTEM_AND_LOGIN';
export const USER_IS_ABSENT_IN_SYSTEM_AND_SIGNUP =
  'app/containers/Button/USER_IS_ABSENT_IN_SYSTEM_AND_SIGNUP';
