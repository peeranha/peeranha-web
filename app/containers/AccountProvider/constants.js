/*
 *
 * AccountProvider constants
 *
 */

export const GET_CURRENT_ACCOUNT = 'app/AccountProvider/GET_CURRENT_ACCOUNT';
export const GET_CURRENT_ACCOUNT_SUCCESS =
  'app/AccountProvider/GET_CURRENT_ACCOUNT_SUCCESS';
export const GET_CURRENT_ACCOUNT_ERROR =
  'app/AccountProvider/GET_CURRENT_ACCOUNT_ERROR';
