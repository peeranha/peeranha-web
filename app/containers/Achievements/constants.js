export const uniqueAchievementsArr = [
  /**
   * FOUNDING_MEMBER does not depends on rating.
   * This is mock values.
   * This approach allows to cover first rating zone
   *  and
   * to avoid unnecessary elements filtering in achievements selectors
   */

  {
    id: 30,
    title: 'FOUNDING_MEMBER',
    lowerValue: -Infinity,
    upperValue: 99,
    limit: 10000,
  },
  {
    id: 31,
    title: 'ACTIVIST',
    lowerValue: 100,
    upperValue: 499,
    limit: process.env.ENV === 'prod' ? 700 : 5,
  },
  {
    id: 32,
    title: 'RESEARCHER',
    lowerValue: 500,
    upperValue: 999,
    limit: process.env.ENV === 'prod' ? 500 : 5,
  },
  {
    id: 33,
    title: 'HONORABLE_RESIDENT',
    lowerValue: 1000,
    upperValue: 2499,
    limit: process.env.ENV === 'prod' ? 300 : 5,
  },
  {
    id: 34,
    title: 'THE_WISEST',
    lowerValue: 2500,
    upperValue: 4999,
    limit: process.env.ENV === 'prod' ? 150 : 5,
  },
  {
    id: 35,
    title: 'MODERN_HERO',
    lowerValue: 5000,
    upperValue: 9999,
    limit: process.env.ENV === 'prod' ? 50 : 5,
  },
  {
    id: 36,
    title: 'KING_OF_THE_HILL',
    lowerValue: 10000,
    upperValue: Infinity,
    limit: 10,
  },
];

export const achievementsArr = [
  /**
   * WELCOME_STRANGER does not depends on rating.
   * This is mock values.
   * This approach allows to cover first rating zone
   *  and
   * to avoid unnecessary elements filtering in achievements selectors
   */

  {
    id: 40,
    title: 'WELCOME_STRANGER',
    lowerValue: -Infinity,
    upperValue: 99,
  },
  { id: 41, title: 'NEWBIE', lowerValue: 100, upperValue: 499 },
  { id: 42, title: 'JUNIOR', lowerValue: 500, upperValue: 999 },
  { id: 43, title: 'RESIDENT', lowerValue: 1000, upperValue: 2499 },
  { id: 44, title: 'SENIOR', lowerValue: 2500, upperValue: 4999 },
  { id: 45, title: 'HERO', lowerValue: 5000, upperValue: 9999 },
  {
    id: 46,
    title: 'SUPERHERO',
    lowerValue: 10000,
    upperValue: Infinity,
  },
];

export const questionsAskedArr = [
  {
    id: 1,
    title: 'QUESTIONS_ASKED_BRONZE',
    lowerValue: process.env.ENV === 'prod' ? 10 : 2,
    upperValue: process.env.ENV === 'prod' ? 49 : 4,
    level: 'bronze',
  },
  {
    id: 2,
    title: 'QUESTIONS_ASKED_SILVER',
    lowerValue: process.env.ENV === 'prod' ? 50 : 5,
    upperValue: process.env.ENV === 'prod' ? 99 : 9,
    level: 'silver',
  },
  {
    id: 3,
    title: 'QUESTIONS_ASKED_GOLD',
    lowerValue: process.env.ENV === 'prod' ? 100 : 10,
    upperValue: Infinity,
    level: 'gold',
  },
];

export const answerGivenArr = [
  {
    id: 10,
    title: 'ANSWERS_GIVEN_BRONZE',
    lowerValue: process.env.ENV === 'prod' ? 20 : 2,
    upperValue: process.env.ENV === 'prod' ? 99 : 4,
    level: 'bronze',
  },
  {
    id: 11,
    title: 'ANSWERS_GIVEN_SILVER',
    lowerValue: process.env.ENV === 'prod' ? 100 : 5,
    upperValue: process.env.ENV === 'prod' ? 199 : 9,
    level: 'silver',
  },
  {
    id: 12,
    title: 'ANSWERS_GIVEN_GOLD',
    lowerValue: process.env.ENV === 'prod' ? 200 : 10,
    upperValue: Infinity,
    level: 'gold',
  },
];

export const bestAnswerArr = [
  {
    id: 20,
    title: 'BEST_ANSWERS_BRONZE',
    lowerValue: process.env.ENV === 'prod' ? 5 : 1,
    upperValue: process.env.ENV === 'prod' ? 24 : 2,
    level: 'bronze',
  },
  {
    id: 21,
    title: 'BEST_ANSWERS_SILVER',
    lowerValue: process.env.ENV === 'prod' ? 25 : 3,
    upperValue: process.env.ENV === 'prod' ? 49 : 4,
    level: 'silver',
  },
  {
    id: 22,
    title: 'BEST_ANSWERS_GOLD',
    lowerValue: process.env.ENV === 'prod' ? 50 : 5,
    upperValue: Infinity,
    level: 'gold',
  },
];

export const firstIn15Arr = [
  {
    id: 50,
    title: 'ANSWERS_IN_15_BRONZE',
    lowerValue: process.env.ENV === 'prod' ? 5 : 1,
    upperValue: process.env.ENV === 'prod' ? 9 : 2,
    level: 'bronze',
  },
  {
    id: 51,
    title: 'ANSWERS_IN_15_SILVER',
    lowerValue: process.env.ENV === 'prod' ? 10 : 3,
    upperValue: process.env.ENV === 'prod' ? 19 : 4,
    level: 'silver',
  },
  {
    id: 52,
    title: 'ANSWERS_IN_15_GOLD',
    lowerValue: process.env.ENV === 'prod' ? 20 : 5,
    upperValue: Infinity,
    level: 'gold',
  },
];

export const firstAnswerArr = [
  {
    id: 60,
    title: 'FIRST_ANSWERS_BRONZE',
    lowerValue: process.env.ENV === 'prod' ? 10 : 1,
    upperValue: process.env.ENV === 'prod' ? 24 : 2,
    level: 'bronze',
  },
  {
    id: 61,
    title: 'FIRST_ANSWERS_SILVER',
    lowerValue: process.env.ENV === 'prod' ? 25 : 3,
    upperValue: process.env.ENV === 'prod' ? 49 : 4,
    level: 'silver',
  },
  {
    id: 62,
    title: 'FIRST_ANSWERS_GOLD',
    lowerValue: process.env.ENV === 'prod' ? 50 : 5,
    upperValue: Infinity,
    level: 'gold',
  },
];

// achievements groups

export const ratingRelated = 'ratingRelated';
export const uniqueRatingRelated = 'uniqueRatingRelated';
export const questionAskedRelated = 'questionAskedRelated';
export const answerGivenRelated = 'answerGivenRelated';
export const bestAnswerRelated = 'bestAnswerRelated';
export const firstAnswerIn15Related = 'firstAnswerIn15Related';
export const firstAnswerRelated = 'firstAnswerRelated';

// action types

const GET_USER_ACHIEVEMENTS = 'containers/Achievements/GET_USER_ACHIEVEMENTS';
const GET_USER_ACHIEVEMENTS_ERROR =
  'containers/Achievements/GET_USER_ACHIEVEMENTS_ERROR';
const GET_USER_ACHIEVEMENTS_SUCCESS =
  'containers/Achievements/GET_USER_ACHIEVEMENTS_SUCCESS';
const SET_VIEW_PROFILE_ACCOUNT =
  'containers/Achievements/SET_VIEW_PROFILE_ACCOUNT';
const RESET_VIEW_PROFILE_ACCOUNT =
  'containers/Achievements/RESET_VIEW_PROFILE_ACCOUNT';
const USER_ACHIEVEMENTS_LOADING =
  'containers/Achievements/USER_ACHIEVEMENTS_LOADING';
const SET_MEMORIZED_ACHIEV_DATA =
  'containers/Achievements/SET_MEMORIZED_ACHIEV_DATA';
const SET_MAX_GROUPS_LOWER_VALUES =
  'containers/Achievements/SET_MAX_GROUPS_LOWER_VALUES';

export {
  GET_USER_ACHIEVEMENTS,
  GET_USER_ACHIEVEMENTS_ERROR,
  GET_USER_ACHIEVEMENTS_SUCCESS,
  SET_VIEW_PROFILE_ACCOUNT,
  RESET_VIEW_PROFILE_ACCOUNT,
  USER_ACHIEVEMENTS_LOADING,
  SET_MEMORIZED_ACHIEV_DATA,
  SET_MAX_GROUPS_LOWER_VALUES,
};
