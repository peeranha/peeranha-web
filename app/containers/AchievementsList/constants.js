export const possibleAchievements = [
  { id: 1, title: 'QUESTION_ASKED' },
  { id: 2, title: 'ANSWER_GIVEN' },
  { id: 3, title: 'CORRECT_ANSWER' },
  { id: 4, title: 'FIRST_10K_REGISTERED' },
  { id: 5, title: 'STRANGER' },
  { id: 6, title: 'NEWBIE' },
  { id: 7, title: 'JUNIOR' },
  { id: 8, title: 'RESIDENT' },
  { id: 9, title: 'SENIOR' },
  { id: 10, title: 'HERO' },
  { id: 11, title: 'SUPERHERO' },
];

const GET_USER_ACHIEVEMENTS =
  'containers/AchievementsList/GET_USER_ACHIEVEMENTS';
const GET_USER_ACHIEVEMENTS_ERROR =
  'containers/AchievementsList/GET_USER_ACHIEVEMENTS_ERROR';
const GET_USER_ACHIEVEMENTS_SUCCESS =
  'containers/AchievementsList/GET_USER_ACHIEVEMENTS_SUCCESS';

export {
  GET_USER_ACHIEVEMENTS,
  GET_USER_ACHIEVEMENTS_ERROR,
  GET_USER_ACHIEVEMENTS_SUCCESS,
};
