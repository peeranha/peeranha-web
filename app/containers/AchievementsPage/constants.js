export const possibleAchievements = [
  { id: 1, title: 'QUESTION_ASKED' },
  { id: 2, title: 'ANSWER_GIVEN' },
  { id: 3, title: 'CORRECT_ANSWER' },
  { id: 4, title: 'FIRST_10K_REGISTERED' },
  { id: 5, title: 'STRANGER' },
  { id: 6, title: 'NEWBIE' },
  { id: 7, title: 'JUNIOR' },
  { id: 8, title: 'RESIDENT' },
  { id: 9, title: 'SENIOR' },
  { id: 10, title: 'HERO' },
  { id: 11, title: 'SUPERHERO' },
];

const GET_USER_ACHIEVEMENTS =
  'containers/AchievementsPage/GET_USER_ACHIEVEMENTS';
const GET_USER_ACHIEVEMENTS_ERROR =
  'containers/AchievementsPage/GET_USER_ACHIEVEMENTS_ERROR';
const GET_USER_ACHIEVEMENTS_SUCCESS =
  'containers/AchievementsPage/GET_USER_ACHIEVEMENTS_SUCCESS';
const SET_CURRENT_ACCOUNT = 'containers/AchievementsPage/SET_CURRENT_ACCOUNT';

export {
  GET_USER_ACHIEVEMENTS,
  GET_USER_ACHIEVEMENTS_ERROR,
  GET_USER_ACHIEVEMENTS_SUCCESS,
  SET_CURRENT_ACCOUNT,
};
