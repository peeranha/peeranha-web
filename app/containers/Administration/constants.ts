const PATH = 'app/Administration/';

export const ADD_MODERATOR_BUTTON_BUTTON =
  'app_Administration_ADD_MODERATOR_BUTTON';
export const OPEN_ADD_MODERATOR_FORM_BUTTON =
  'app_Administration_OPEN_ADD_MODERATOR_FORM_BUTTON';

export const WALLET_ADDRESS_FIELD = `${PATH}WALLET_ADDRESS_FIELD`;

export const GET_MODERATORS = `${PATH}GET_MODERATORS`;
export const GET_MODERATORS_SUCCESS = `${PATH}GET_MODERATORS_SUCCESS`;
export const GET_MODERATORS_ERROR = `${PATH}GET_MODERATORS_ERROR`;

export const ADD_MODERATOR = `${PATH}ADD_MODERATOR`;
export const ADD_MODERATOR_SUCCESS = `${PATH}ADD_MODERATOR_SUCCESS`;
export const ADD_MODERATOR_ERROR = `${PATH}ADD_MODERATOR_ERROR`;

export const REVOKE_MODERATOR = `${PATH}REVOKE_MODERATOR`;
export const REVOKE_MODERATOR_SUCCESS = `${PATH}REVOKE_MODERATOR_SUCCESS`;
export const REVOKE_MODERATOR_ERROR = `${PATH}REVOKE_MODERATOR_ERROR`;
