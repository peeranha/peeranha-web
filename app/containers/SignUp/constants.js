/*
 *
 * SignUp constants
 *
 */

// Fields

const EOS_NAME_FIELD = 'eos_name';
const EMAIL_FIELD = 'email';
const VERIFICATION_FIELD = 'verification';
const EOS_ACTIVE_PRIVATE_KEY_FIELD = 'eos_active_private_key';
const EOS_OWNER_PRIVATE_KEY_FIELD = 'eos_owner_private_key';
const STORE_KEY_FIELD = 'store_key';
const MASTER_KEY_FIELD = 'master_key';
const DISPLAY_NAME_FIELD = 'display_name';
const TELOS_NAME_FIELD = 'telos_name';
const MY_OWN_TELOS_NAME_FIELD = 'my_own_telos_name';
const PASSWORD_FIELD = 'password';
const PASSWORD_CONFIRM_FIELD = 'password_confirm';
const I_SAVE_MASTER_KEY_FIELD = 'i_save_master_key';
const I_ACCEPT_PRIVACY_POLICY_FIELD = 'i_accept_privacy_policy';
const WHY_DO_YOU_LIKE_US_FIELD = 'why_do_you_like_us';
const EOS_ACCOUNT_FIELD = 'eos_account';

// Action types

const SET_REDUCER_DEFAULT = 'app/SignUp/SET_REDUCER_DEFAULT';

const EMAIL_CHECKING = 'app/SignUp/EMAIL_CHECKING';
const EMAIL_CHECKING_SUCCESS = 'app/SignUp/EMAIL_CHECKING_SUCCESS';
const EMAIL_CHECKING_ERROR = 'app/SignUp/EMAIL_CHECKING_ERROR';

const EMAIL_VERIFICATION = 'app/SignUp/EMAIL_VERIFICATION';
const EMAIL_VERIFICATION_SUCCESS = 'app/SignUp/EMAIL_VERIFICATION_SUCCESS';
const EMAIL_VERIFICATION_ERROR = 'app/SignUp/EMAIL_VERIFICATION_ERROR';
const SEND_ANOTHER_CODE = 'app/SignUp/SEND_ANOTHER_CODE';

const I_HAVE_EOS_ACCOUNT = 'app/SignUp/I_HAVE_EOS_ACCOUNT';
const I_HAVE_EOS_ACCOUNT_SUCCESS = 'app/SignUp/I_HAVE_EOS_ACCOUNT_SUCCESS';
const I_HAVE_EOS_ACCOUNT_ERROR = 'app/SignUp/I_HAVE_EOS_ACCOUNT_ERROR';

const I_HAVE_NOT_EOS_ACCOUNT = 'app/SignUp/I_HAVE_NOT_EOS_ACCOUNT';
const I_HAVE_NOT_EOS_ACCOUNT_SUCCESS =
  'app/SignUp/I_HAVE_NOT_EOS_ACCOUNT_SUCCESS';
const I_HAVE_NOT_EOS_ACCOUNT_ERROR = 'app/SignUp/I_HAVE_NOT_EOS_ACCOUNT_ERROR';

const SIGNUP_WITH_WALLET = 'app/SignUp/SIGNUP_WITH_WALLET';
const SIGNUP_WITH_WALLET_SUCCESS = 'app/SignUp/SIGNUP_WITH_WALLET_SUCCESS';
const SIGNUP_WITH_WALLET_ERROR = 'app/SignUp/SIGNUP_WITH_WALLET_ERROR';
const SIGNUP_WITH_WALLET_REFERRAL_ERROR =
  'app/SignUp/SIGNUP_WITH_WALLET_REFERRAL_ERROR';

const SHOW_WALLET_SIGNUP_FORM = 'app/SignUp/SHOW_WALLET_SIGNUP_FORM';
const SHOW_WALLET_SIGNUP_FORM_SUCCESS =
  'app/SignUp/SHOW_WALLET_SIGNUP_FORM_SUCCESS';
const SHOW_WALLET_SIGNUP_FORM_ERROR =
  'app/SignUp/SHOW_WALLET_SIGNUP_FORM_ERROR';

const PUT_KEYS_TO_STATE = 'app/SignUp/PUT_KEYS_TO_STATE';

// Errors

const USER_ALREADY_REGISTERED_ERROR = 'USER_ALREADY_REGISTERED';

// Other
const ACCOUNT_NOT_CREATED_NAME = 'peeranhanone';
const AUTOGENERATED = 'autogenerated';
const MY_OWN = 'my_own';

export {
  SET_REDUCER_DEFAULT,
  EMAIL_CHECKING,
  EMAIL_CHECKING_SUCCESS,
  EMAIL_CHECKING_ERROR,
  EOS_NAME_FIELD,
  EMAIL_FIELD,
  VERIFICATION_FIELD,
  EOS_ACTIVE_PRIVATE_KEY_FIELD,
  EOS_OWNER_PRIVATE_KEY_FIELD,
  STORE_KEY_FIELD,
  MASTER_KEY_FIELD,
  DISPLAY_NAME_FIELD,
  TELOS_NAME_FIELD,
  MY_OWN_TELOS_NAME_FIELD,
  PASSWORD_FIELD,
  PASSWORD_CONFIRM_FIELD,
  I_SAVE_MASTER_KEY_FIELD,
  WHY_DO_YOU_LIKE_US_FIELD,
  EOS_ACCOUNT_FIELD,
  I_ACCEPT_PRIVACY_POLICY_FIELD,
  EMAIL_VERIFICATION,
  EMAIL_VERIFICATION_SUCCESS,
  EMAIL_VERIFICATION_ERROR,
  I_HAVE_EOS_ACCOUNT,
  I_HAVE_EOS_ACCOUNT_SUCCESS,
  I_HAVE_EOS_ACCOUNT_ERROR,
  I_HAVE_NOT_EOS_ACCOUNT,
  I_HAVE_NOT_EOS_ACCOUNT_SUCCESS,
  I_HAVE_NOT_EOS_ACCOUNT_ERROR,
  PUT_KEYS_TO_STATE,
  SIGNUP_WITH_WALLET,
  SIGNUP_WITH_WALLET_SUCCESS,
  SIGNUP_WITH_WALLET_ERROR,
  SIGNUP_WITH_WALLET_REFERRAL_ERROR,
  SHOW_WALLET_SIGNUP_FORM,
  SHOW_WALLET_SIGNUP_FORM_SUCCESS,
  SHOW_WALLET_SIGNUP_FORM_ERROR,
  USER_ALREADY_REGISTERED_ERROR,
  ACCOUNT_NOT_CREATED_NAME,
  SEND_ANOTHER_CODE,
  AUTOGENERATED,
  MY_OWN,
};
