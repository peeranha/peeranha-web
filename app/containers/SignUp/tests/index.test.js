import { fromJS } from 'immutable';
import createdHistory from 'createdHistory';
import * as routes from 'routes-config';

import { generateKeys } from 'utils/web_integration/src/util/eos-keygen';
import { generateMasterKey } from 'utils/web_integration/src/util/masterKeygen';

import { SignUp } from '../index';
import { EMAIL_FIELD } from '../constants';

jest.mock('utils/web_integration/src/util/eos-keygen', () => ({
  generateKeys: jest.fn(),
}));

jest.mock('utils/web_integration/src/util/masterKeygen', () => ({
  generateMasterKey: jest.fn(),
}));

jest.mock('createdHistory', () => ({
  push: jest.fn(),
}));

window.URL = {
  createObjectURL: jest.fn(),
};

const cmp = new SignUp();

beforeEach(() => {
  createdHistory.push.mockClear();
  generateKeys.mockClear();
  generateMasterKey.mockClear();
  window.URL.createObjectURL.mockClear();

  cmp.props = {
    locale: 'en',
    children: jest.fn(),
    showLoginModalDispatch: jest.fn(),
    checkEmailDispatch: jest.fn(),
    verifyEmailDispatch: jest.fn(),
    iHaveEosAccountDispatch: jest.fn(),
    idontHaveEosAccountDispatch: jest.fn(),
    signUpWithScatterDispatch: jest.fn(),
    emailChecking: false,
    emailVerificationProcessing: false,
    iHaveEosAccountProcessing: false,
    idontHaveEosAccountProcessing: false,
    signUpWithScatterProcessing: false,
    showScatterSignUpProcessing: false,
    showScatterSignUpFormDispatch: jest.fn(),
    account: 'account',
    email: 'email',
    withScatter: false,
    keys: {},
    putKeysToStateDispatch: jest.fn(),
  };
});

describe('SignUp', () => {
  it('render', () => {
    expect(cmp.render()).toMatchSnapshot();
  });

  it('getLinkToDownloadKeys', () => {
    const keys = { key1: 'key1', key2: 'key2' };

    const text = JSON.stringify({ keys });
    const data = new Blob([text], { type: 'text/plain' });

    cmp.getLinkToDownloadKeys();

    expect(window.URL.createObjectURL).toHaveBeenCalledWith(data);
  });

  it('getAllKeys', async () => {
    const linkToDownloadAllKeys = 'getAllKeys';
    const keys = { activeKey: 1, ownerKey: 2 };

    generateKeys.mockImplementation(() => keys);
    window.URL.createObjectURL.mockImplementationOnce(
      () => linkToDownloadAllKeys,
    );

    await cmp.getAllKeys();

    expect(cmp.props.putKeysToStateDispatch).toHaveBeenCalledWith({
      activeKey: keys.activeKey,
      ownerKey: keys.ownerKey,
      linkToDownloadAllKeys,
    });
  });

  it('getMasterKey', () => {
    const masterKey = 'masterKey';
    const linkToDownloadMasterKey = 'getMasterKey';

    generateMasterKey.mockImplementation(() => masterKey);
    window.URL.createObjectURL.mockImplementationOnce(
      () => linkToDownloadMasterKey,
    );

    cmp.getMasterKey();

    expect(cmp.props.putKeysToStateDispatch).toHaveBeenCalledWith({
      masterKey,
      linkToDownloadMasterKey,
    });
  });

  describe('checkEmail', () => {
    it('with values', () => {
      const values = fromJS({
        [EMAIL_FIELD]: EMAIL_FIELD,
      });

      expect(cmp.props.checkEmailDispatch).toHaveBeenCalledTimes(0);
      cmp.checkEmail(values);

      expect(cmp.props.checkEmailDispatch).toHaveBeenCalledTimes(1);
      expect(cmp.props.checkEmailDispatch).toHaveBeenCalledWith(
        values.get(EMAIL_FIELD),
      );
    });

    it('without values', () => {
      const event = {};

      expect(cmp.props.checkEmailDispatch).toHaveBeenCalledTimes(0);
      cmp.checkEmail(event);

      expect(cmp.props.checkEmailDispatch).toHaveBeenCalledTimes(1);
      expect(cmp.props.checkEmailDispatch).toHaveBeenCalledWith(
        cmp.props.email,
      );
    });
  });

  describe('componentWillMount', () => {
    it('without @email in store', () => {
      cmp.props.email = null;
      cmp.props.withScatter = false;

      expect(createdHistory.push).toHaveBeenCalledTimes(0);
      cmp.componentWillMount();

      expect(createdHistory.push).toHaveBeenCalledTimes(1);
      expect(createdHistory.push).toHaveBeenCalledWith(
        routes.signup.email.name,
      );
    });

    it('with @email in store', () => {
      cmp.props.email = 'email';
      cmp.props.withScatter = true;

      expect(createdHistory.push).toHaveBeenCalledTimes(0);
      cmp.componentWillMount();
      expect(createdHistory.push).toHaveBeenCalledTimes(0);
    });

    it('without @keys in store', () => {
      cmp.props.keys = null;

      expect(cmp.props.putKeysToStateDispatch).toHaveBeenCalledTimes(0);
      cmp.componentWillMount();
      expect(cmp.props.putKeysToStateDispatch).toHaveBeenCalled();
    });

    it('with @keys in store', () => {
      cmp.props.keys = {};

      expect(cmp.props.putKeysToStateDispatch).toHaveBeenCalledTimes(0);
      cmp.componentWillMount();
      expect(cmp.props.putKeysToStateDispatch).toHaveBeenCalledTimes(0);
    });
  });
});
