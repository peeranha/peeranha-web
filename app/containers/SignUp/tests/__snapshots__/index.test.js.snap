// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp snapshot render testing case 1: content === SHOW_DEFAULT_SIGNUP_MODAL 1`] = `
<ModalDialog
  closeModal={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  show={true}
>
  <div>
    <SignUpOptions
      backToOptions={[Function]}
      continueSignUp={[Function]}
    />
  </div>
</ModalDialog>
`;

exports[`SignUp snapshot render testing case 2: content === USER_IS_ABSENT_IN_SYSTEM_AND_SIGNUP 1`] = `
<ModalDialog
  closeModal={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  show={true}
>
  <div>
    <ReduxForm
      account="user"
      errorMessage={null}
      loading={false}
      registerUser={[Function]}
      translations={
        Object {
          "app.components.Footer.header": "Copyright: Peerania",
          "app.components.Header.about": "About",
          "app.components.Header.help": "Help",
          "app.components.Header.home": "Home",
          "app.components.Header.login": "Log in",
          "app.components.Header.profile": "Profile",
          "app.components.Header.search": "Search",
          "app.components.Header.signUp": "Sign Up",
          "app.components.NoAccess.errorMessage": "No access to complete this action!",
          "app.containers.ErrorBoundary.description": "Description of error boundary",
          "app.containers.ErrorBoundary.details": "Details",
          "app.containers.ErrorBoundary.problemWithWebpage": "A problem occurred with this webpage",
          "app.containers.ErrorBoundary.reloadPage": "Reload page",
          "app.containers.ErrorBoundary.title": "Error",
          "app.containers.Login.doNotHaveAcc": "Do not have an account?",
          "app.containers.Login.header": "No Peerania account exists for selected identity.",
          "app.containers.Login.loginWithScatter": "Log in with Scatter",
          "app.containers.Login.requirementToSignUp": "Sign Up for Peerania account",
          "app.containers.Login.selectAnotherIdentity": "Select another identity",
          "app.containers.Login.signUp": "Sign up",
          "app.containers.Profile.aboutLabel": "About me",
          "app.containers.Profile.avatarLabel": "Avatar",
          "app.containers.Profile.cancelButton": "Cancel",
          "app.containers.Profile.chooseFile": "Choose file",
          "app.containers.Profile.companyLabel": "Company",
          "app.containers.Profile.displayNameLabel": "Display name",
          "app.containers.Profile.editButton": "Edit",
          "app.containers.Profile.emptyField": "This field is empty",
          "app.containers.Profile.fileSize": "File size is exceeded",
          "app.containers.Profile.isProfileLoading": "Loading...",
          "app.containers.Profile.locationLabel": "Location",
          "app.containers.Profile.positionLabel": "Position",
          "app.containers.Profile.profile": "Profile",
          "app.containers.Profile.profileDescription": "Profile description",
          "app.containers.Profile.saveButton": "Save",
          "app.containers.Profile.savingButton": "Saving...",
          "app.containers.Profile.stringLength": "String is too short or too long",
          "app.containers.Profile.viewButton": "View",
          "app.containers.Profile.wrongUser": "Wrong user",
          "app.containers.ScatterInstaller.backToOptions": "Back to options",
          "app.containers.ScatterInstaller.header": "To complete this action in our app you need to install",
          "app.containers.ScatterInstaller.nextstep": "Next step",
          "app.containers.SelectAccount.backToOptions": "Back to options",
          "app.containers.SelectAccount.header": "Scatter account was not selected.",
          "app.containers.SelectAccount.selectIdent": "Select identity",
          "app.containers.SignUp.alreadyHaveAcc": "Already have Peerania account?",
          "app.containers.SignUp.backToSignUpOptions": "Back to Sign Up options",
          "app.containers.SignUp.displayName": "Display Name",
          "app.containers.SignUp.displayNameLength": "Display name length must be between 3 and 20",
          "app.containers.SignUp.eosAccount": "EOS Account",
          "app.containers.SignUp.identity": "identity",
          "app.containers.SignUp.isAlreadyRegistred": "is already registred",
          "app.containers.SignUp.login": "Log in",
          "app.containers.SignUp.loginWith": "Login with",
          "app.containers.SignUp.or": "or",
          "app.containers.SignUp.requiredField": "Required field",
          "app.containers.SignUp.selectedIdentity": "Selected identity",
          "app.containers.SignUp.serverMessage": "Error",
          "app.containers.SignUp.signUp": "Sign Up",
          "app.containers.SignUp.signUpDescription": "Description of Sign Up",
          "app.containers.SignUp.signUpWith": "Sign up with Scatter",
          "app.containers.UserProfileNav.profile": "Profile",
        }
      }
    />
  </div>
</ModalDialog>
`;

exports[`SignUp snapshot render testing case 3: content === USER_IS_IN_SYSTEM_AND_SIGNUP 1`] = `
<ModalDialog
  closeModal={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  show={true}
>
  <div>
    <IdentityIsRegistred
      account="user"
      backToOptions={[Function]}
      continueLogin={[Function]}
    />
  </div>
</ModalDialog>
`;

exports[`SignUp snapshot render testing case 4: content === NO_SELECTED_SCATTER_ACCOUNTS 1`] = `
<ModalDialog
  closeModal={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  show={true}
>
  <div>
    <SelectAccountComponent
      backToOptions={[Function]}
      selectAccount={[Function]}
    />
  </div>
</ModalDialog>
`;

exports[`SignUp snapshot render testing case 5: content === NO_SCATTER 1`] = `
<ModalDialog
  closeModal={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  show={true}
>
  <div>
    <ScatterInstaller
      backToOptions={[Function]}
    />
  </div>
</ModalDialog>
`;
